packages/scesim-editor/src/creation/TestScenarioCreationPanel.tsx

```
@@ -37,45 +39,142 @@ import CubesIcon from "@patternfly/react-icons/dist/esm/icons/cubes-icon";
+                basename(modelA).localeCompare(basename(modelB))
+              )
+            );
+          })
+          .catch((err) => {
+            setCallBackError(err);
+            console.error(
+              `[TestScenarioCreationPanel] The below error when trying to retrieve all the External DMN files from the project.`
+            );
+            console.error(err);
+          });
+      },
+      [onRequestExternalModelsAvailableToInclude]
+    )
+  );
+
+  /** This callback return the unmarshalled representation of a DMN model given its path */
+  useCancelableEffect(
+    useCallback(
+      ({ canceled }) => {
+        if (!selectedDmnModelPathRelativeToThisScesim || onRequestExternalModelByPath === undefined) {
+          return;
+        }
+
+        onRequestExternalModelByPath(selectedDmnModelPathRelativeToThisScesim)
+          .then((externalDMNModel) => {
+            console.trace("[TestScenarioCreationPanel] The below external DMN model have been loaded");
+            console.trace(externalDMNModel);
+
+            if (canceled.get() || !externalDMNModel) {
+              return;
+            }
+
+            setSelectedDmnModel(externalDMNModel);
+          })
+          .catch((err) => {
+            setCallBackError(err);
+            console.error(
+              `[TestScenarioCreationPanel] An error occurred when parsing the selected model '${selectedDmnModelPathRelativeToThisScesim}'. Please double-check it is a non-empty valid model.`
+            );
+            console.error(err);
+          });
+      },
+      [onRequestExternalModelByPath, selectedDmnModelPathRelativeToThisScesim]
+    )
+  );
+
+  /* If any error occurs during the execution of useCancelableEffect's callback,    */
+  /* it throws the error that will be catched by the ErrorBoundary.                 */
+  useEffect(() => {
+    if (callBackError) {
+      throw callBackError;
+    }
+  }, [callBackError]);
```

If we're letting error propagete, why do we need the `callBackError` state? Why not let them explode when they happen? Mapping errors as state is usually for a try/catch-like mechanism, when we want the screen to show a different UI for when errors occur.

---

Good observation, so it's enough to rethrow the error inside the catch block, right?

---

What try/catch block? Sorry it seems that are multiple ones. If an error happens and we do something on the screen to "treat it", I think we shouldn't re-throw it. Not if it's really an "end of the world" type of error, I think we should just throw and let the first [ErrorBoundary](https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary) catch it.

---

>  Mapping errors as state is usually for a try/catch-like mechanism, when we want the screen to show a different UI for when errors occur.

I guess that is the case. The error is retrieved and set in the state in the catch blocks at lines 92 and 124. 
Without that useEffect(), the error is NOT propagated to the error boundary, at least with the tests I did.